src/wifipaf/tests/TestWiFiPAFTP.cpp

```diff
@@ -0,0 +1,180 @@
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <cstdint>
+#include <cstring>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+#include <pw_unit_test/framework.h>
+
+#include <lib/core/CHIPError.h>
+#include <lib/core/StringBuilderAdapters.h>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/Span.h>
+#include <lib/support/TypeTraits.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <system/SystemLayer.h>
+#include <system/SystemPacketBuffer.h>
+
+#include <wifipaf/WiFiPAFTP.h>
+
+namespace chip {
+namespace WiFiPAF {
+
+class TestWiFiPAFTP : public WiFiPAFTP, public ::testing::Test
+{
+public:
+    static void SetUpTestSuite()
+    {
+        ASSERT_EQ(chip::Platform::MemoryInit(), CHIP_NO_ERROR);
+        ASSERT_EQ(DeviceLayer::SystemLayer().Init(), CHIP_NO_ERROR);
+    }
+
+    static void TearDownTestSuite()
+    {
+        DeviceLayer::SystemLayer().Shutdown();
+        chip::Platform::MemoryShutdown();
+    }
+
+    void SetUp() override { ASSERT_EQ(Init(nullptr, false), CHIP_NO_ERROR); }
+
+    void TearDown() override {}
+};
+
+TEST_F(TestWiFiPAFTP, CheckLogState)
+{
+    LogState();
+    TakeTxPacket();
+    TakeRxPacket();
+
+    PacketBufferHandle AckToSend = System::PacketBufferHandle::New(kTransferProtocolStandaloneAckHeaderSize);
+    EXPECT_EQ(EncodeStandAloneAck(AckToSend), CHIP_NO_ERROR);
+
+    EXPECT_NE(GetAndIncrementNextTxSeqNum(), 0);
+    EXPECT_NE(GetAndRecordRxAckSeqNum(), 0);
+    HasUnackedData();
+
+    uint16_t FragSize = 300;
+    SetTxFragmentSize(FragSize);
+    SetRxFragmentSize(FragSize);
+    EXPECT_EQ(GetTxFragmentSize(), FragSize);
+    EXPECT_EQ(GetRxFragmentSize(), FragSize);
+
+    EXPECT_NE(GetRxNextSeqNum(), 0);
+    EXPECT_NE(GetLastReceivedSequenceNumber(), 0);
+    EXPECT_NE(GetNewestUnackedSentSequenceNumber(), 0);
+    EXPECT_EQ(ExpectingAck(), true);
+    ClearRxPacket();
+    ClearTxPacket();
+}
+
+TEST_F(TestWiFiPAFTP, CheckSendSingle)
+{
+    auto buf = System::PacketBufferHandle::New(100);
+    buf->SetDataLength(100);
+    mRxNextSeqNum = 1;
+    EXPECT_TRUE(HandleCharacteristicSend(buf.Retain(), true));
+    EXPECT_EQ(TxState(), kState_Complete);
+    TakeTxPacket();
+    EXPECT_EQ(TxState(), kState_Idle);
+}
+
+TEST_F(TestWiFiPAFTP, CheckSendMultiple)
+{
+    auto buf = System::PacketBufferHandle::New(500);
+    buf->SetDataLength(500);
```

Comment:
This buffer is random data from the stack, so when you allocate, please clear them otherwise some static analyzers may complain that data is used uninitialized.

Reply:
I will add to clear the buffer. Thanks.